app/controllers/api/v2/hosts_controller.rb

```diff
@@ -331,15 +331,15 @@ def host_attributes(params, host = nil)
           end
           # map interface types
           params[:interfaces_attributes] = params[:interfaces_attributes].map do |nic_attr|
-            interface_attributes(nic_attr)
+            interface_attributes(nic_attr, skip_nil_type: !host.nil?)
           end
         end
         params = host.apply_inherited_attributes(params) if host
         params
       end
 
-      def interface_attributes(params)
-        params[:type] = InterfaceTypeMapper.map(params[:type])
+      def interface_attributes(params, skip_nil_type: false)
+        params[:type] = InterfaceTypeMapper.map(params[:type]) unless skip_nil_type && !params.has_key?(:type)
```

Comment:
The condition translates to `if !skip_nil_type || params.has_key?(:type)` which means, that the the line would be skipped, if the `skip_nil_type` is set, which is not the same as if the `param[:type]` was nil. It doesn't break the behavior, but might cause issues later. Changing the order to `if params.has_key?(:type) || !skip_nil_type` (+ using if instead of unless here, as it's hard to reason about double negation in complex conditions - at least for me). Or even better would be having `allow_nil_type` with condition  `if params.has_key?(:type) || allow_nil_type`